File: Java/Animal.java
interface Animal {
    void fazerSom();
    void mover();
}

class Cachorro implements Animal {
    @Override
    public void fazerSom() {
        System.out.println("O cachorro faz: Au Au!");
    }

    @Override
    public void mover() {
        System.out.println("O cachorro corre!");
    }
}

class Gato implements Animal {
    @Override
    public void fazerSom() {
        System.out.println("O gato faz: Miau!");
    }

    @Override
    public void mover() {
        System.out.println("O gato pula!");
    }
}

File: Kotlin/Animal.kt
interface Animal {
    fun fazerSom()
    fun mover()
}

class Cachorro : Animal {
    override fun fazerSom() {
        println("Cachorro faz: Au Au")
    }

    override fun mover() {
        println("Cachorro está correndo")
    }
}

class Gato : Animal {
    override fun fazerSom() {
        println("Gato faz: Miau")
    }

    override fun mover() {
        println("Gato está caminhando silenciosamente")
    }
}
